### PR TITLE
Design for OSC containerization

### PR DESCRIPTION
**Pull request by arvindn05**

### CODE REVIEW

File: designs/osc_containerization/OSC-containerization.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+##### Life cycle management
+Remove restart and shutdown operations possbile via OSC UI
+Remove inplace upgrade of OSC via server upgrade bundles
+
+##### CLI/Restricted Shell
+Remove restricted shell. Restricted shell is no longer applicable as we will be using standard base containers.
+Note: It is technically feasible to have a restricted shell if we build up the OSC container image from scratch. For now, this is out of scope.
+
+##### Network/Ip address management
```
Comment (sudhirappaji) [inline designs/osc_containerization/OSC-containerization.md]: I presume we can still run all the scenarios of OSC (like creation VC,MC,DA,DS, SG binding etc) with changes proposed in the networking/ip addressing management.

File: designs/osc_containerization/OSC-containerization.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+###### Scope
+- Create docker files for OSC
+- Use Docker volume to store persistent data and decouple from OSC code
+	- Moved log files to persistent volume
+	- Moved Database to persisent volume
+	- Moved Plugins to persistent volume
+- Launch different OSC containers and make sure they are using the same persistent data
+
+Code is available [here](https://github.com/arvindn05/osc-core/tree/docker)
```
Comment (sudhirappaji) [inline designs/osc_containerization/OSC-containerization.md]: I see in the code, that OSC can run only in console mode and no UI? Can we have more details?

File: designs/osc_containerization/OSC-containerization.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+N/A
+
+### OSC Synchronization Tasks
+N/A
+
+## Tests
+N/A
+
+## References
+### [Dockerfile reference](https://docs.docker.com/engine/reference/builder/)
```
Comment (sudhirappaji) [inline designs/osc_containerization/OSC-containerization.md]: Can you add more references and we don't have to search for additional information elsewhere?

File: designs/osc_containerization/OSC-containerization.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+It also simplifies packaging and distribution.
+
+## Constraints and Assumptions
+
+##### Constraint
+- Support migrating current OSC installation to container based installation.
+
+##### Assumption
+- OSC will be deployed as a container only. The Virtual Machine appliance model will not be supported going forward.
```
[the remark this answers is not in the document]
Comment (arvindn05) [reply designs/osc_containerization/OSC-containerization.md]: I have advised them as such....we havent received strong requirements either way. For now the scope is to investigate the fastest approach.

File: designs/osc_containerization/OSC-containerization.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+- For migration of VM-based OSC to container based OSC, some manual steps maybe required. Scripts should be provided in this case.
```
[the remark this answers is not in the document]
Comment (arvindn05) [reply designs/osc_containerization/OSC-containerization.md]: Until we actually do the migration i am not sure. I would assume copy the database/keystore to the modified location etc. Will add the above as an example

File: designs/osc_containerization/OSC-containerization.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+- Plugins
+- Service Function catalog image files
+- Log file
+- vmidcServer.conf
+
+This data would need to be exposed in docker volumes to the OSC container.
+
+##### Life cycle management
+Remove restart and shutdown operations possbile via OSC UI
+Remove inplace upgrade of OSC via server upgrade bundles
```
[the remark this answers is not in the document]
Comment (arvindn05) [reply designs/osc_containerization/OSC-containerization.md]: good point will update the UI impact.

For the second point, see assumption.

File: designs/osc_containerization/OSC-containerization.md (hunk 1 above)
Comment (arvindn05) [reply designs/osc_containerization/OSC-containerization.md]: yes.

File: designs/osc_containerization/OSC-containerization.md (hunk 2 above)
Comment (arvindn05) [reply designs/osc_containerization/OSC-containerization.md]: UI is still available. Console mode refers to the starting of OSC.

File: designs/osc_containerization/OSC-containerization.md (hunk 3 above)
Comment (arvindn05) [reply designs/osc_containerization/OSC-containerization.md]: Any examples of what other information you want here? A lot of this functionality is documented as part of docker or other tutorials online.

**Comment by arvindn05** [comment]
Please let me know of any addtional comments and i will try to integrate.

**Pull request merged**